packages/main/src/ColorPicker.js

```diff
@@ -0,0 +1,531 @@
+		}
+
+		this._changeSelectedColor(x, y);
+		this.mouseIn = false;
+		this.mouseDown = false;
+	}
+
+	_handleMouseMove(event) {
+		if (!this.mouseDown || !this.mouseIn) {
+			return;
+		}
+
+		this._changeSelectedColor(event.offsetX, event.offsetY);
+	}
+
+	_handleAlphaInput(event) {
+		this._alpha = parseFloat(event.target.value);
+	}
+
+	_handleHueInput(event) {
+		this._hue = event.target.value;
+		this._setMainColor(this._hue);
+	}
+
+	_handleHEXChange(event) {
+		const newValue = event.target.value.toLowerCase();
+		const hexRegex = new RegExp("^[<0-9 abcdef]+$");
+
+		this.hex = newValue;
+		if (newValue.length !== 6 || !hexRegex.test(newValue)) {
+			this._wrongHEX = true;
+		} else {
+			this._wrongHEX = false;
+			this._color = this._hexToRGB(this.hex);
+			this._setValuesFromHEX();
+		}
+	}
+
+	_handleRGBInputsChange(event) {
+		const targetValue = parseInt(event.target.value);
+
+		switch (event.target.id) {
+		case "red":
+			this._color = Object.assign({ }, this._color, { r: targetValue });
+			break;
+
+		case "green":
+			this._color = Object.assign({ }, this._color, { g: targetValue });
+			break;
+
+		case "blue":
+			this._color = Object.assign({ }, this._color, { b: targetValue });
+			break;
+		}
+
+		this._setHex();
+		this._setValuesFromHEX();
+	}
+
+	_setMainColor(hueValue) {
+		if (hueValue <= 255) {
+			this._mainColor = {
+				r: 255,
+				g: hueValue,
+				b: 0,
+			};
+		} else if (hueValue <= 510) {
+			this._mainColor = {
+				r: 255 - (hueValue - 255),
+				g: 255,
+				b: 0,
+			};
+		} else if (hueValue <= 765) {
+			this._mainColor = {
+				r: 0,
+				g: 255,
+				b: hueValue - 510,
+			};
+		} else if (hueValue <= 1020) {
+			this._mainColor = {
+				r: 0,
+				g: 765 - (hueValue - 255),
+				b: 255,
+			};
+		} else if (hueValue <= 1275) {
+			this._mainColor = {
+				r: hueValue - 1020,
+				g: 0,
+				b: 255,
+			};
+		} else {
+			this._mainColor = {
+				r: 255,
+				g: 0,
+				b: 1275 - (hueValue - 255),
+			};
+		}
+	}
+
+	_handleAlphaChange(event) {
+		this._alpha = this._alpha < 0 ? 0 : this._alpha;
+		this._alpha = this._alpha > 1 ? 1 : this._alpha;
+	}
+
+	_changeSelectedColor(x, y) {
+		this._selectedCoordinates = {
+			x: x - 6.5, // Center the coordinates, because of the width of the circle
+			y: y - 6.5, // Center the coordinates, because of the height of the circle
+		};
+
+		this._setColor(x, y);
+		this._setHex();
+	}
+
+	_setColor(x, y) {
+		// By using the selected coordinates(x = Lightness, y = Saturation) and hue(selected from the hue slider)
+		// and HSL format, the color will be parsed to RGB
+
+		const h = Math.round(this._hue / 4.25), // 0 ≤ H < 360
+			s = 1 - (Math.round(((y / 256) + Number.EPSILON) * 100) / 100), // 0 ≤ S ≤ 1
+			l = Math.round(((x / 256) + Number.EPSILON) * 100) / 100; // 0 ≤ V ≤ 1
+
+		if (!s || !l) {
+			// The event is finished out of the main color section
+			return;
+		}
+
+		this._color = this._HSLToRGB({ h, s, l });
+	}
+
+	_HSLToRGB(color = {
```

Comment:
Functions that don't work with a class's state don't belong there. Please move this one and the others similar to it to some utils/ diretory as separate modules